dra-evolution/pkg/api/poc.yaml

```diff
@@ -0,0 +1,287 @@
+sharedAttributeGroups:
+- name: common-attributes
+  attributes:
+  - name: product-name
+    string: Mock NVIDIA A100-SXM4-40GB
+  - name: brand
+    string: Nvidia
+  - name: architecture
+    string: Ampere
+  - name: cuda-compute-capability
+    version: 8.0.0
+  - name: driver-version
+    version: 550.54.15
+  - name: cuda-driver-version
+    version: 12.4.0
+
+sharedCapacityTemplates:
+- name: gpu-shared-resources
+  capacities:
+  - name: multiprocessors
+    quantity: "98"
+  - name: copy-engines
+    quantity: "7"
+  - name: decoders
+    quantity: "5"
+  - name: encoders
+    quantity: "0"
+  - name: jpeg-engines
+    quantity: "1"
+  - name: ofa-engines
+    quantity: "1"
+  - name: memory-slices
+    intRange: 0-7
+
+deviceTemplates:
+- name: gpu
+  sharedAttributeGroups:
+  - common-attributes
+  attributes:
+  - name: memory
+    quantity: 40Gi
+  - name: mig-capable
+    bool: true
+  sharedCapacitiesConsumed:
+  - sharedCapacityTemplateName: gpu-shared-resources
+    capacities:
+    - name: multiprocessors
+      quantity: "98"
+    - name: copy-engines
+      quantity: "7"
+    - name: decoders
+      quantity: "5"
+    - name: encoders
+      quantity: "0"
+    - name: jpeg-engines
+      quantity: "1"
+    - name: ofa-engines
+      quantity: "1"
+    - name: memory-slices
+      intRange: 0-7
+- name: mig-1g.5gb
+  sharedAttributeGroups:
+  - common-attributes
+  attributes:
+  - name: mig-profile
+    string: 1g.5gb
+  - name: memory
+    quantity: 4864Mi
+  sharedCapacitiesConsumed:
+  - sharedCapacityTemplateName: gpu-shared-resources
+    capacities:
+    - name: multiprocessors
+      quantity: "14"
+    - name: copy-engines
+      quantity: "1"
+    - name: decoders
+      quantity: "0"
+    - name: encoders
+      quantity: "0"
+    - name: jpeg-engines
+      quantity: "0"
+    - name: ofa-engines
+      quantity: "0"
+
+devices:
+- name: gpu
+  namespace: gpu-0
+  deviceTemplateName: gpu
+  attributes:
+  - name: index
+    string: "0"
+  - name: minor
+    string: "0"
+  - name: uuid
+    string: GPU-0eaad900-5263-4fd6-b020-f03d30efac31
+- name: mig-1g.5gb-0
+  namespace: gpu-0
+  deviceTemplateName: mig-1g.5gb
+  attributes:
+  - name: index
+    string: "0:0"
+  - name: parentUUID
+    string: GPU-4404041a-04cf-1ccf-9e70-f139a9b1e23c
+  sharedCapacitiesConsumed:
+  - sharedCapacityTemplateName: gpu-shared-resources
+    capacities:
+    - name: memory-slices
```

Comment:
So these capacities override the or, rather, are in addition to the ones listed in the deviceTemplate?

In other words, this saying:
- the `sharedCapacityConsumed` comes from the named sharedCapacityTemplate ("group" seems like a better word than template?) in the namespace `gpu-0`
- the values should be the ones from the `deviceTemplate` and also the ones from here

Reply:
Yes. But in my latest iteration I have actually created separate deviceTemplates for these to avoid having to include it here.